patches/server/0010-Adventure.patch

```diff
@@ -2288,11 +2309,18 @@ index 49138ccda0f378b13c7f425be765876eb4026b06..ddc91d376fd59da982e594188fcef720
 +            @Override
 +            public <T> DataResult<T> encode(final Component input, final DynamicOps<T> ops, final T prefix) {
 +                if (input instanceof io.papermc.paper.adventure.AdventureComponent adv) {
-+                    if (adv.deepConvertedIfPresent() != null) {
-+                        return origCodec.encode(java.util.Objects.requireNonNull(adv.deepConvertedIfPresent()), ops, prefix);
-+                    } else {
-+                        // return io.papermc.paper.adventure.PaperAdventure.localizedCodec(locale).encode(adv.adventure$component(), ops, prefix); // TODO
-+                        return io.papermc.paper.adventure.AdventureCodecs.COMPONENT_CODEC.encode(adv.adventure$component(), ops, prefix);
++                    return io.papermc.paper.adventure.PaperAdventure.localizedCodec(locale)
++                        .encode(adv.adventure$component(), ops, prefix);
++                }
++                if (locale != null && input.getContents() instanceof TranslatableContents) {
++                    final net.kyori.adventure.text.Component adventureComponent
++                        = io.papermc.paper.adventure.PaperAdventure.asAdventure(input);
```

Comment:
I think once we convert to adventure all we need to do is pass the result into the relevant adventure Codec, like is done above. Then we don't need to make #translated public either.